lib/charms/vault_k8s/v0/juju_facade.py

```diff
@@ -330,6 +349,14 @@ def grant_secret(
         secret = self.get_secret(id=secret_id, label=secret_label)
         secret.grant(relation)
 
+    def remove_secret(self, label: str, id: str | None = None) -> None:
+        """Remove the secret if it exists."""
+        try:
+            secret = self.get_secret(label=label, id=id)
+            secret.remove_all_revisions()
+        except (NoSuchSecretError, SecretNotFoundError):
+            return
```

Comment:
I would avoid not doing anything when there are exceptions, I believe we should at least log something.